src/execution/compiler/codegen.cpp

```diff
@@ -429,15 +428,13 @@ ast::Expr *CodeGen::PeekValue(const parser::ConstantValueExpression &const_val)
     }
     case type::TypeId::DATE: {
       const auto val = const_val.Peek<execution::sql::Date>().ToNative();
-      auto ymd = terrier::util::TimeConvertor::YMDFromDate(static_cast<type::date_t>(val));
-      auto year = static_cast<int32_t>(ymd.year());
-      auto month = static_cast<uint32_t>(ymd.month());
-      auto day = static_cast<uint32_t>(ymd.day());
+      int32_t year = 0, month = 0, day = 0;
+      execution::sql::Date::FromNative(val).ExtractComponents(&year, &month, &day);
       return DateToSql(year, month, day);
     }
     case type::TypeId::TIMESTAMP: {
       const auto val = const_val.Peek<execution::sql::Timestamp>().ToNative();
-      auto julian_usec = terrier::util::TimeConvertor::ExtractJulianMicroseconds(static_cast<type::timestamp_t>(val));
+      auto julian_usec = execution::sql::Timestamp::FromNative(val).ToNative();
```

Comment:
All of the ToNative FromNative in a consecutive lines is confusing. Is there an abstraction (like, just a function) with a descriptive name and documentation that takes an execution timestamp to Julian usec? That seems to be what all of these steps accomplish.

Reply:
Alternatively, just some documentation of the steps being performed is good too.

Reply:
Right, I lost that I'm not using the timestamp_t type anymore in translation, so just using ToNative should suffice as the value represents the Julian microseconds in the first place -- updated accordingly.